doc/contributor/howto-guide-adding-generated-libraries.md

```diff
@@ -28,26 +28,31 @@ cd $HOME/google-cloud-cpp
 ```shell
 library=... # The name of your new library in the google-cloud-cpp repository
 subdir="google/cloud/${library}"  # The path in googleapis repo, may not start with google/cloud/
-bazel_output_base="$(bazel info output_base)"
+bazel_output_base="$(bazelisk info output_base)"
 ```
 
 ### Verify the C++ rules exist
 
+You may need to
+[Send a PR to update the googleapis SHA to the latest version](../contributor/howto-guide-update-googleapis-sha.md).
+Wait until that is submitted before proceeding any further for the following 2
+cases:
+
+1. The dependency does not exist at the pinned version of the googleapis repo.
+
 ```shell
-bazel --batch query --noshow_progress --noshow_loading_progress \
+bazelisk --batch query --noshow_progress --noshow_loading_progress \
     "kind(cc_library, @com_google_googleapis//${subdir}/...)"
 ```
 
-If the command fails, it returns something like this:
+- If the command fails, it returns something like this:
 
 ```shell
 ERROR: no targets found beneath 'commerce'
 ```
 
-This means the dependency does not exist at the pinned version of the googleapis
-repo.
-[Send a PR to update the googleapis SHA to the latest version](../contributor/howto-guide-update-googleapis-sha.md).
-Wait until that is submitted before proceeding any further.
+2. Check `$bazel_output_base/external/googleapis~/api-index-v1.json`, if
```

Comment:
Isn't this the same case?

Reply:
When I generate `parallelstore`, the command return no error, but I failed in running the library generation, because no parallelstore.v1 in api-index-v1.json. I have to make #14801 first.

Reply:
Oh, ok. Yeah, you're right. The api-index-v1.json is updated weekly, independent of any new services.

https://github.com/googleapis/googleapis/commits/master/api-index-v1.json